### PR TITLE
Fix chat

### PR DESCRIPTION
**Pull request by oversword**
Depends on https://github.com/minetest-mods/teleport-request/pull/45
Depends on https://github.com/minetest-mods/teleport-request/pull/46

Fixes duplication of the message code and the constant checking for `chat2`

Also adds the ability to customise the chat color by a setting (requests are so easy to miss on a busy server)

There are a few instances where ONLY buit-in chat OR chat2 were used... I'm unsure why it would be this way, and assumed it should always be sent to both - if this is wrong @Panquesito7 please let me know and I can change it back

Will need to test the new color setting and check messages are sending properly

### CODE REVIEW

File: settingtypes.txt [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Timeout delay (default is 60)
 tp.timeout_delay (Timeout delay after a request is sent) int 60
 
+# Message color (default is #FFFFFF)
+tp.message_color (The color of the messages sent to players by this mod) string #FFFFFF
```
Comment (Panquesito7) [inline settingtypes.txt:5]:
```suggestion
tp.message_color (The color of the messages sent to players) string #FFFFFF
```

**Review by Panquesito7 — APPROVED** [review]
Thanks so much for your hard work! I'll merge this PR in a few days unless there are any objections. 🙂

**Comment by oversword** [comment]
> Thanks so much for your hard work!

You're welcome, thanks for the mod(s)!

File: settingtypes.txt [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Timeout delay (default is 60)
 tp.timeout_delay (Timeout delay after a request is sent) int 60
 
+# Message color (default is #FFFFFF)
```
Comment (Panquesito7) [inline settingtypes.txt:4]:
```suggestion
# Message color (default color is #FFFFFF)
```

**Pull request merged**